### PR TITLE
feat: Deprecate `datasets` module, rename to `inferences`

### LINKED ISSUE #2732: [ENHANCEMENT] rename Dataset to Inferences
Need to carve out room for future datasets. We should deprecate the old datasets and rename them

### PR DESCRIPTION
**Pull request by anticorrelator**
Resolves #2732 

- Deprecates `phoenix.datasets` module, interfaces are available under `phoenix.inferences`
- `px.Dataset` class renamed to `px.Inference`
- `px.ExampleDatasets` class renmaed to `px.ExampleInferences`
- `Datasets.from_open_inference` deprecated and removed from `Inference`

### CODE REVIEW

**Review by RogerHYang — APPROVED** [review]

File: tests/server/api/types/test_dataset.py
Comment (axiomofjoy) [inline tests/server/api/types/test_dataset.py:5]: Rename to `InternalInference`.

File: src/phoenix/core/model.py
Comment (axiomofjoy) [inline src/phoenix/core/model.py:9]: I think we can rename these arguments for clarity.

File: src/phoenix/core/model_schema_adapter.py
Comment (axiomofjoy) [inline src/phoenix/core/model_schema_adapter.py:136]: These probably deserve to be renamed.

File: src/phoenix/server/main.py
Comment (axiomofjoy) [inline src/phoenix/server/main.py:117]: Probably can be renamed.

File: tests/datasets/test_inference.py
Comment (axiomofjoy) [inline tests/datasets/test_inference.py:25]: Looks like the test names in this file are still named with `dataset` names.

File: tests/server/api/types/test_inference.py
Comment (axiomofjoy) [inline tests/server/api/types/test_inference.py:38]: looks like this file still has a lot of dataset names

File: src/phoenix/inferences/fixtures.py
Comment (axiomofjoy) [inline src/phoenix/inferences/fixtures.py:416]: This can be renamed.

File: src/phoenix/inferences/fixtures.py
Comment (axiomofjoy) [inline src/phoenix/inferences/fixtures.py:421]: same here

File: src/phoenix/inferences/inference.py
Comment (axiomofjoy) [inline src/phoenix/inferences/inference.py:37]: Should this be plural because it represents multiple inferences?

**Review by axiomofjoy — APPROVED** [review]
`Inference` vs. `Inferences`?

Just checking the diff, I'm still seeing a large number of old `dataset` names.

**Review by mikeldking — CHANGES_REQUESTED** [review]
I think it should be plural. Also can we cascade the notebook changes? We need to wait until the release. You will also have to pin the notebooks lowerbound

**Review by mikeldking — APPROVED** [review]
Approving to unblock

**Pull request merged**
